livekit-plugins/livekit-plugins-openai/livekit/plugins/openai/realtime/realtime_model.py

```diff
@@ -767,6 +810,82 @@ def session_update(
             }
         )
 
+    async def aitem_create(
+        self,
+        message: llm.ChatMessage,
+        previous_item_id: str | None = None,
+        _on_create_callback: Callable[[], None] | None = None,
+    ) -> None:
+        fut = asyncio.Future[None]()
+        self._item_created_futs[message.id] = fut
+        self.conversation.item.create(message, previous_item_id)
+        if _on_create_callback:
+            _on_create_callback()
+        await fut
+        del self._item_created_futs[message.id]
+
+    async def aitem_delete(self, item_id: str) -> None:
+        fut = asyncio.Future[None]()
+        self._item_deleted_futs[item_id] = fut
+        self.conversation.item.delete(item_id)
+        await fut
+        del self._item_deleted_futs[item_id]
+
+    async def async_chat_ctx(self, new_ctx: llm.ChatContext) -> None:
```

Comment:
I was more thinking about making the current API async. Feel free to break the API. The RealtimeModel API is still beta :)

Reply:
Do you mean to make the `self.conversation.item.create` and `self.conversation.item.delete` be async? 

Maybe I can move the `acreate` and `adelete` to the `conversation.item` but keep the original version of create and delete, as for some cases like `sync_chat_ctx` we may want to add all the create and delete messages to the event queue at once and wait at the end?

Reply:
> Do you mean to make the self.conversation.item.create and self.conversation.item.delete be async?
Yep.

I don't think we need two functions (sync and async)

Reply:
My concern is that when we call `gather(item.acreate(item1), item.create(item2))`, is there a guarantee for the two items are pushed to the message queue in the correct order?